### PR TITLE
FINERACT-2494: Add unit tests for ApiParameterHelper in fineract-core.

### PR DESCRIPTION
**Pull request by San-43**
## Description

This PR fixes [FINERACT-2494](https://issues.apache.org/jira/browse/FINERACT-2494?jql=project%20%3D%20%22Apache%20Fineract%22%20AND%20labels%20%3D%20gsoc-fineract-evidence%20ORDER%20BY%20createdDate).

**New test coverage for API parameter extraction and parsing:**

* Added `ApiParameterHelperTest` with tests for extracting and validating `commandId`, fields, associations, and locale from query parameters, including handling of missing, blank, invalid, and duplicate values.
* Included tests for boolean query parameter parsing for methods such as `parameterType`, `template`, `makerCheckerable`, `includeJson`, and `genericResultSet`, verifying correct handling of true, false, and missing values.
* Provided tests for exclusion logic in associations and for presence checks on the `genericResultSet` flag.
* Added a helper method to simplify test parameter creation, improving readability and maintainability of the test code.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per [our guidelines](https://github.com/apache/fineract/blob/develop/CONTRIBUTING.md#pull-requests)
- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.
- [x] Create/update [unit or integration tests](https://fineract.apache.org/docs/current/#_testing) for verifying the changes made.
- [x] Follow our [coding conventions](https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions).
- [x] [This PR must not be a "code dump"](https://cwiki.apache.org/confluence/display/FINERACT/Pull+Request+Size+Limit). Large changes can be made in a branch, with assistance. Ask for help on the [developer mailing list](https://fineract.apache.org/#contribute).

Your assigned reviewer(s) will follow our [guidelines for code reviews](https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide).

### CODE REVIEW

**Comment by Ambika-Sony** [comment]
Hi @San-43 , I see you're working on this issue!

I’ve already submitted PR #5665 addressing FINERACT-2494. My approach focuses on building a comprehensive unit test suite for ApiParameterHelper, including coverage for edge cases such as null handling, negative offsets, malformed inputs, and boundary conditions.

The tests are designed as isolated JUnit 5 unit tests to ensure reliability and maintainability within fineract-core, and I’m also aligning with static analysis improvements from FINERACT-2510.

Happy to align approaches or incorporate any feedback from the community.

**Comment by San-43** [comment]
Hi @Ambika-Sony, thanks for the note and for your work on this. I actually hadn’t noticed #5665 when I started #5677. I saw a couple of closed PR targeting [FINERACT-2494](https://issues.apache.org/jira/browse/FINERACT-2494?jql=project%20%3D%20%22Apache%20Fineract%22%20AND%20labels%20%3D%20gsoc-fineract-evidence%20ORDER%20BY%20createdDate), but I missed #5665, apologies for that.

After comparing #5665 and #5677, my view is that #5677 is a bit cleaner for FINERACT-2494 because it keeps the change strictly tests-only, in line with the ticket scope. In #5665, there are also `ApiParameterHelper` production-code changes; those hardening updates are valuable, but they may be better handled in a separate PR/ticket so scope and review impact stay clear.

I’m happy to align and avoid duplication. If maintainers prefer a single path, I can either:

1. close this PR, or
2. keep this one for tests-only and leave behavior changes to a follow-up PR.

Happy to follow maintainer guidance either way.

**Comment by Ambika-Sony** [comment]
> Hi @Ambika-Sony, thanks for the note and for your work on this. I actually hadn’t noticed #5665 when I started #5677. I saw a couple of closed PR targeting [FINERACT-2494](https://issues.apache.org/jira/browse/FINERACT-2494?jql=project%20%3D%20%22Apache%20Fineract%22%20AND%20labels%20%3D%20gsoc-fineract-evidence%20ORDER%20BY%20createdDate), but I missed #5665, apologies for that.
> 
> After comparing #5665 and #5677, my view is that #5677 is a bit cleaner for [FINERACT-2494](https://issues.apache.org/jira/browse/FINERACT-2494) because it keeps the change strictly tests-only, in line with the ticket scope. In #5665, there are also `ApiParameterHelper` production-code changes; those hardening updates are valuable, but they may be better handled in a separate PR/ticket so scope and review impact stay clear.
> 
> I’m happy to align and avoid duplication. If maintainers prefer a single path, I can either:
> 
> 1. close this PR, or
> 2. keep this one for tests-only and leave behavior changes to a follow-up PR.
> 
> Happy to follow maintainer guidance either way.

Hi @San-43,

Regarding the scope: the 'production-code changes' in #5665 weren't just extra additions; they are hardening fixes specifically required to address NullPointerExceptions and boundary errors that my 22+ test cases exposed. A tests-only PR (#5677) would pass on the current code but would leave these underlying stability issues unaddressed.

I believe it's critical for the project to not only have tests but to resolve the vulnerabilities those tests reveal. I'm happy to let the maintainers decide the best path forward, but I will be keeping #5665 active as a comprehensive solution for FINERACT-2494 that ensures both coverage and core stability.

File: fineract-core/src/test/java/org/apache/fineract/infrastructure/core/api/ApiParameterHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import org.apache.fineract.infrastructure.core.exception.PlatformApiDataValidationException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class ApiParameterHelperTest {
+
+    @Test
```
Comment (Aman-Mittal) [inline fineract-core/src/test/java/org/apache/fineract/infrastructure/core/api/ApiParameterHelperTest.java:49]: You can manintain Constants in this file for duplicated string literals

Comment (San-43) [reply fineract-core/src/test/java/org/apache/fineract/infrastructure/core/api/ApiParameterHelperTest.java:49]: Thanks, @Aman-Mittal. I've added the constants and rebased onto the latest develop. Pls let me know if there is anything else I should add/change.
